source.lua

```diff
@@ -1,38 +1,68 @@
 --@ module = true
 local repeatUtil = require('repeat-util')
+local json = require('json')
+local persist = require('persist-table')
 
-liquidSources = liquidSources or {}
+local GLOBAL_KEY = 'source' -- used for state change hooks and persistence
 
 local sourceId = 'liquidSources'
 
+g_sources_list = g_sources_list or {}
+local function retrieve_state()
+    return g_sources_list
+end
+
+local function has_elements(collection)
+    for _,_ in pairs(collection) do return true end
+    return false
+end
```

Comment:
Since I wrote `prioritize` I found a better way to write this. instead of `has_elements(t)` you can write `next(t)` and take advantage of the built-in `next` keyword